lib/Transport.js

```diff
@@ -610,8 +610,9 @@ function isStream (obj) {
 }
 
 function defaultNodeFilter (node) {
-  // avoid master only nodes
-  if (node.roles.master === true &&
+  // avoid cluster_manager or master only nodes
+  // toDo: remove role check on master when master is not supported
+  if (node.roles.cluster_manager === true | node.roles.master === true &&
```

Comment:
might be easier to read if:
``((nodes.roles.cluster_manager === true || node.roles.master === true))``